core/tee/tee_svc_cryp.c

```diff
@@ -3527,6 +3508,20 @@ TEE_Result syscall_asymm_verify(unsigned long state,
 		break;
 
 	case TEE_MAIN_ALGO_DSA:
+		hash_algo = TEE_DIGEST_HASH_TO_ALGO(cs->algo);
+		res = tee_hash_get_digest_size(hash_algo, &hash_size);
+		if (res != TEE_SUCCESS)
+			goto out;
```

Comment:
Since we're in a `switch()` we can break on error instead, this is done in some cases but not all here.
I'd recommend breaking on error instead of `goto out`.